### ISSUE TITLE
More robust documentation

### ISSUE THREAD

**Issue by EricKuck** [bluelinelabs/Conductor#186]
Maybe a wiki?

**Comment by sockeqwe** [comment]
As far as I know, github pages are now generated automatically from markdown files that are placed in a folder `/docs`

https://github.com/blog/2289-publishing-with-github-pages-now-as-easy-as-1-2-3
https://github.com/blog/2228-simpler-github-pages-publishing

I think a simple github page would be a great place for a documentation, and since it's on the master branch we could add a link on each documentation page to edit that file directly in github which will result in a pull request when pressing the save button.

If you think that this is too much work and not worth the initial setup effort, than github wiki would be my first alternative. I, personally, would prefer a github page over a wiki

**Comment by hannesstruss** [comment]
I agree with @sockeqwe. Keeping docs and code in one place is like the better way. For example, it allows for better control over whether pull requests to the code contain the necessary doc updates.

**Comment by nomisRev** [comment]
Any plans? I just spend a couple of hours figuring out why I had a memory leak in ControllerPagerAdapter because I was not setting the adapter of ViewPager back to null `onDestroyView` and a couple of days ago that the `Router` returned by a Controller hosted in the ViewPager was returning a `ChildRouter` of ViewPager instead of the ActivityHostedRouter.

Just an idea, that could be useful to document. I definitely want to help out with some docs.

**Comment by EricKuck** [comment]
I'll have more time to get started on this coming up pretty soon. I'm not awesome at documentation though and was hoping for some suggestions.

- How should they be organized? By class? By feature? By use case?
- Where do the little "gotchas" go? For example, @nomisRev's issue with setting the `ViewPager`'s adapter to null in `onDestroyView`?
- What's the best way to collaborate on this? I'd love whatever help I can get, but am not sure how to delegate for this.

**Comment by hannesstruss** [comment]
I think a build of the JavaDoc + something like [OkHttp recipes](https://github.com/square/okhttp/wiki/Recipes) would be great.
